### ISSUE TITLE
Can someone verify that this crashes? (interp)

### ISSUE THREAD

**Issue by jeanlaroche** [serge-sans-paille/pythran#1319]
This code is giving me a core dumped on the lastest master, on LINUX, python3
```
import numpy as np

#pythran export mel_p()
def mel_p():
    mels = np.array([#pythran export mel_p()
def mel_p():
    mels = np.array([3,4.22619,5.45238,6.67857,7.90476,9.13095,10.3571,11.5833,3,4.22619,5.45238,6.67857,7.90476,9.13095,10.3571,11.5833])
    mel_f = np.interp(mels[0:10], np.array([0., 15., 20., 35., 40., 41.]), np.array([0., 200., 1000., 13000., 24000., 96000.]))
    print("TEST")
    return mel_f])
    mel_f = np.interp(mels[0:10], np.array([0., 15., 20., 35., 40., 41.]), np.array([0., 200., 1000., 13000., 24000., 96000.]))
    print("TEST")
    return mel_f
```
When called with
```
python -c "from foo import * ; mel_p()"
```
The code works fine on MacOS, so I'm really puzzled here. Can someone repro the problem?
Puzzling: "TEST" gets printed.
I'm not sure how to debug this.
Compilation pastebin:
https://pastebin.com/jcniagZD

**Comment by jeanlaroche** [comment]
I believe I found the root cause of the crash, a bad allocation of a temp array in interp_core. It would help, though if someone else could verify both the problem and the fix...
The issue was declaring a vector in a scope, and using its pointer outside of the declaration scope, so it's not fully determined what happens in that case (what's sure is that's a terrible idea).

**Comment by serge-sans-paille** [comment]
I agree with your #1320 fix, provided you apply the small suggestion (and squash it)

**Comment by serge-sans-paille** [comment]
Fixed by #1320
